locator/network_topology_strategy.cc

```diff
@@ -265,12 +265,17 @@ network_topology_strategy::calculate_natural_endpoints(
     co_return std::move(tracker.replicas());
 }
 
-void network_topology_strategy::validate_options(const gms::feature_service& fs) const {
+void network_topology_strategy::validate_options(const gms::feature_service& fs, const locator::topology& topology) const {
     if(_config_options.empty()) {
         throw exceptions::configuration_exception("Configuration for at least one datacenter must be present");
     }
+    auto dcs = topology.get_datacenters();
     validate_tablet_options(*this, fs, _config_options);
     for (auto& c : _config_options) {
+        if (!dcs.contains(c.first)) {
```

Comment:
That's effectively the essence of this patch, which is lost in propagating topology over the stack. I'd split this patch into two -- "propagate topology" and "this change" -- and then merge "this change" with previous patch (titled `cql: remove incorrect tags check in ...`)

Reply:
I tried splitting/merging this commit in a couple of different ways but I was not satisfied with any result. My conclusion is: this change belongs here, it justifies why we need the extra `topology` argument. Also, the commit description says: 
> `validate_options` needs to be extended with `topology` parameter, because NetworkTopologyStrategy needs to validate if every explicitly listed DC is really existing.

So this `if` is emphasized in the description. 
Let me know if you still want to split it.

Reply:
Things have changed since the comment has been made, please re-evaluate the comment.